metrics/api/src/main/java/io/helidon/metrics/api/DistributionStatisticsConfig.java

```diff
@@ -0,0 +1,204 @@
+/*
+ * Copyright (c) 2023 Oracle and/or its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.helidon.metrics.api;
+
+import java.time.Duration;
+
+/**
+ * Configuration which controls the behavior of distribution statistics from meters that support them
+ * (for example, timers and distribution summaries).
+ *
+ */
+public interface DistributionStatisticsConfig extends Wrapped {
+
+    /**
+     * Returns whether the configuration is set for percentile histograms which can be aggregated for percentile approximations.
+     *
+     * @return whether percentile histograms are configured
+     */
+    boolean isPercentileHistogram();
+
+    /**
+     * Returns whether the configuration is set to publish percentiles.
+     *
+     * @return true/false
+     */
+    boolean isPublishingPercentiles();
+
+    /**
+     * Returns whether the configuration is set to publish a histogram.
+     *
+     * @return true/false
+     */
+    boolean isPublishingHistogram();
+
+    /**
+     * Returns the settings for non-aggregable percentiles.
+     *
+     * @return percentiles to compute and publish
+     */
+    Iterable<Double> percentiles();
+
+    /**
+     * Returns the configured number of digits of precision for percentiles.
+     *
+     * @return digits of precision to maintain for percentile approximations
+     */
+    int percentilePrecision();
+
+    /**
+     * Returns the maximum value that the meter is expected to observe.
+     *
+     * @return maximum value that the meter is expected to observe
+     */
+    double maximumExpectedValue();
+
+    /**
+     * Returns how long decaying past observations remain in the ring buffer.
+     *
+     * @see #bufferLength()
+     * @return time during which samples accumulate in a histogram
+     */
+    Duration expiry();
+
+    /**
+     * Returns the size of the ring buffer for holding decaying observations.
+     *
+     * @return number of observations to keep in the ring buffer
+     */
+    int bufferLength();
+
+    /**
+     * Returns the configured service level objective boundaries.
+     *
+     * @return the SLO boundaries
+     */
+    Iterable<Double> serviceLevelObjectiveBoundaries();
+
+    /**
+     * Builder for a new {@link io.helidon.metrics.api.DistributionStatisticsConfig} instance.
+     */
+    interface Builder extends io.helidon.common.Builder<Builder, DistributionStatisticsConfig> {
```

Comment:
Please use the new builder approach - e.g. `DistributionStatisticsConfigBlueprint` interface instead of custom builder

Reply:
@tomas-langer I think we could use the builder generator; I'll be interested to see if you think it is worth it, because...

This is not a traditional config and builder--they are wrappers around their counterparts in an implementation of the metrics API. This includes an `unwrap` method which returns a cast reference to the delegate from the implementation library. The `Wrapped` interface declares `unwrap` and the hand-written `DistributionStatisticsConfig` interface currently extends `Wrapped`. Then each implementation implements the config interface and its inner builder interface. Quite straightforward.

I can create a blueprint for this interface, but if I add `@Prototype.Implement("io.helidon.metrics.api.Wrapped")` to it then the generated config implementation class fails to compile because it lacks an implementation of `unwrap`. The actual  implementation of that method depends on which implementation library is in use. If I could have the generator mark the generated config implementation class as `abstract` then this would work smoothly. But I cannot do that AFAIK.

One way we _could_ use the builder generator would be this (just for clarity using Micrometer as the implementation):
1. The config blueprint would be `DistributionStatisticsConfigInternalBlueprint` with methods declared for the various settings as usual.
2. Hand-write    
   ```
   public interface DistributionStatisticsConfig extends DistributionStatisticsConfigInternal, Wrapped {
   
         abstract class Builder extends DistributionStatisticsConfigInternal.BuilderBase<Builder, DistributionStatisticsConfig> {
         }
       }
   ```
   
   That's all. I'm not leaving anything out.
3. Each adapter implementation (e.g., `helidon-metrics-micrometer`) would implement `DistributionStatisticsConfig` and its `Builder` would extend `DistributionStatisticsConfig.Builder` to implement `buildPrototype` which would 
   1. Instantiate the Micrometer distribution stats config builder.
   2. Use the inherited, generated methods on the our `DistributionStatisticsConfig` (inherited from the generated internal interface) to fetch all the settings and assign those values on the Micrometer config builder.
   3. Invoke the Micrometer `builder.build()` method to get the Micrometer config instance.
   4. Instantiate our Micrometer adapter's implementation class for `DistributionStatisticsConfig` with its delegate set to the Micrometer config object.

Unfortunately, the generated "internal" interface would be public and we really would rather hide it. There is also an extra level of inheritance we have to introduce.

So, because this is not a traditional config type and trying to use the generator causes some issues, maybe it's OK to hand-code this config type and its builder.

Reply:
ok, let's do it for now